WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -417,3 +439,83 @@ extension OrderDetailsViewModel {
         stores.dispatch(deleteTrackingAction)
     }
 }
+
+extension OrderDetailsViewModel: OrderDetailsViewModelStatusSubModel {
+
+    // TODO
+    // configureResultsController
+    //X statusResultsController.startForwardingEvents(to: tableView)
+
+    func refreshStatuses() {
+        do {
+            try statusResultsController.performFetch()
+        } catch {
+            CrashLogging.logError(error)
+        }
+    }
+
+    func getNumStatusSections() -> Int {
+        return statusResultsController.sections.count
+    }
+
+    func getNumStatuses(inSection section: Int) -> Int {
+        return statusResultsController.sections[section].numberOfObjects
+    }
+
+    func getStatusName(at indexPath: IndexPath) -> String? {
+        let status = statusResultsController.object(at: indexPath)
+        return status.name
+    }
+
+    func getIndexOfSelectedStatus() -> IndexPath? {
+        guard let orderStatus = self.orderStatus else {
+            return nil
+        }
+        guard let row = statusResultsController.fetchedObjects.firstIndex(of: orderStatus) else {
+            return nil
+        }
+        return IndexPath(row: row, section: 0)
+    }
+
+    func setSelectedStatus(to indexPath: IndexPath) {
+        let selectedStatus = statusResultsController.object(at: indexPath)
+        let newStatus = selectedStatus.status
+        let done = updateOrderAction(siteID: order.siteID, orderID: order.orderID, status: newStatus)
+        ServiceLocator.stores.dispatch(done)
+
+        //X TODO hook up undo, notice
+
+        //X let undo = updateOrderAction(siteID: order.siteID, orderID: orderID, status: undoStatus)
+
+        //X ServiceLocator.analytics.track(.orderStatusChange,
+        //X                           withProperties: ["id": orderID,
+        //X                                            "from": undoStatus.rawValue,
+        //X                                            "to": newStatus.rawValue])
+
+        //X displayOrderUpdatedNotice {
+        //X     ServiceLocator.stores.dispatch(undo)
+        //X     ServiceLocator.analytics.track(.orderStatusChange,
+        //X                               withProperties: ["id": orderID,
+        //X                                                "from": newStatus.rawValue,
+        //X                                                "to": undoStatus.rawValue])
+        //X }
+    }
+
+    /// Returns an Order Update Action that will result in the specified Order Status updated accordingly.
+    ///
+    private func updateOrderAction(siteID: Int64, orderID: Int64, status: OrderStatusEnum) -> Action {
+        return OrderAction.updateOrder(siteID: siteID, orderID: orderID, status: status, onCompletion: { error in
+            guard let error = error else {
```

Comment:
nit: Following the [golden path rule](https://www.raywenderlich.com/1179-swift-style-guide-april-2016-update#:~:text=The%20Golden%20Path%20(or%20sometimes,quickly%20gets%20hard%20to%20read.&text=If%20you%20can%20help%20it%2C%20don't%20nest%20at%20all.), I would suggest putting your error handling code inside the `if` branch and do your success path outside it.

```swift
if let error = error {
    // handle error
    return 
}

// handle success
```

Reply:
I was surprised by this too. It was in the code that way already, and is present elsewhere https://github.com/woocommerce/woocommerce-ios/search?q=%22guard+let+error%22

If you'd like me to change it nonetheless, I am happy to do so